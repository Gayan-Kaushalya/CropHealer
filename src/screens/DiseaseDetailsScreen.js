import { StyleSheet, Text, View, Image, SafeAreaView, Pressable, ScrollView } from "react-native";
import React from "react";
import { FontAwesome } from "@expo/vector-icons";
import BackButton from "../components/BackButton";

const DiseaseDetailsScreen = ({ route }) => {
    const { disease } = route.params || {};

    //console.log("Route Params:", route.params);
    //console.log("Disease:", disease);

    return (
        <ScrollView style={{ backgroundColor: disease.color, flex: 1 }} showsVerticalScrollIndicator={false}>
            <BackButton/>
            <View
                style={{
                    backgroundColor: "#fff",
                    flex: 1,
                    marginTop: 240,
                    borderTopLeftRadius: 56,
                    borderTopRightRadius: 56,
                    alignItems: "center",
                    paddingHorizontal: 16,
                }}>
                <View style={{
                    height: 300,
                    width: 300,
                    position: "absolute",
                    top: -150,
                }}>
                    <Image source={disease.image} style={{ height: "100%", width: "100%", resizeMode: "contain" }} />
                </View>

                {/* Disease Name */}
                <Text style={{ fontSize: 24, fontWeight: "bold", marginTop: 160 }}>{disease.disease}</Text>

                {/* Description */}    
                <Text style={{ marginHorizontal: 16, textAlign: "center", marginTop: 16 }}>{disease.description}</Text>

                {/* Extra Details */}
                <View style={{ alignSelf: "flex-start", marginHorizontal: 22, marginBottom: 16, width: '100%' }}>
                    <Text style={{ fontSize: 18, fontWeight: "bold", marginTop: 16 }}>Crops:</Text>
                    <Text style={{ marginLeft: 16 }}>{disease.crops.join(", ")}</Text>

                    <Text style={{ fontSize: 18, fontWeight: "bold", marginTop: 16 }}>Symptoms:</Text>
                    {disease.symptoms.map((item, index) => (
                        <View key={index} style={{ flexDirection: 'row', alignItems: 'flex-start', marginVertical: 2, marginLeft: 16 }}>
                            <Text style={{ marginRight: 8 }}>{'\u2022'}</Text>
                            <Text style={{ flex: 1 }}>{item}</Text>
                        </View>
                    ))}

                    <Text style={{ fontSize: 18, fontWeight: "bold", marginTop: 16 }}>Causes:</Text>
                    {disease.causes.map((item, index) => (
                        <View key={index} style={{ flexDirection: 'row', alignItems: 'flex-start', marginVertical: 2, marginLeft: 16 }}>
                            <Text style={{ marginRight: 8 }}>{'\u2022'}</Text>
                            <Text style={{ flex: 1 }}>{item}</Text>
                        </View>
                    ))}

                    <Text style={{ fontSize: 18, fontWeight: "bold", marginTop: 16 }}>Cures:</Text>
                    {disease.cures.map((item, index) => (
                        <View key={index} style={{ flexDirection: 'row', alignItems: 'flex-start', marginVertical: 2, marginLeft: 16 }}>
                            <Text style={{ marginRight: 8 }}>{'\u2022'}</Text>
                            <Text style={{ flex: 1 }}>{item}</Text>
                        </View>
                    ))}

                    <Text style={{ fontSize: 18, fontWeight: "bold", marginTop: 16 }}>Preventive Measures:</Text>
                    {disease.preventive_measures.map((item, index) => (
                        <View key={index} style={{ flexDirection: 'row', alignItems: 'flex-start', marginVertical: 2, marginLeft: 16 }}>
                            <Text style={{ marginRight: 8 }}>{'\u2022'}</Text>
                            <Text style={{ flex: 1 }}>{item}</Text>
                        </View>
                    ))}
                </View>
            </View>
        </ScrollView>
    );
};

export default DiseaseDetailsScreen;

const styles = StyleSheet.create({});