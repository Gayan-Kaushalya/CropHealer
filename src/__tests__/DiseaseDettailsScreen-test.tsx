import React from "react";
import { render } from "@testing-library/react-native";
import DiseaseDetailsScreen from "../screens/DiseaseDetailsScreen"; // Adjust the path as necessary
import { NavigationContainer } from "@react-navigation/native";

// Sample disease data for testing
const mockDisease = {
  disease: "Bacterial Spot",
  description: "Bacterial Spot is a common disease in plants.",
  color: "#ffcccb",
  image: require("../../assets/diseases/bacterial_spot.png"), // Adjust the path as needed
  crops: ["Tomato", "Pepper", "Eggplant"],
  symptoms: ["Leaf spots", "Wilting", "Fruit rot"],
  causes: ["Bacteria", "Poor drainage"],
  cures: ["Remove infected plants", "Use resistant varieties"],
  preventive_measures: ["Proper irrigation", "Crop rotation"],
};

describe("<DiseaseDetailsScreen />", () => {
  it("renders correctly with given disease data", () => {
    const { toJSON } = render(
      <NavigationContainer>
        <DiseaseDetailsScreen route={{ params: { disease: mockDisease } }} />
      </NavigationContainer>
    );

    // Create a snapshot of the rendered component
    expect(toJSON()).toMatchSnapshot();
  });
});
