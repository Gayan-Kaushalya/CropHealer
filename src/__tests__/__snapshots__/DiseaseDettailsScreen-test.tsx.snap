// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiseaseDetailsScreen /> renders correctly with given disease data 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    {
      "backgroundColor": "#ffcccb",
      "flex": 1,
    }
  }
>
  <View>
    <RCTSafeAreaView
      style={
        {
          "flexDirection": "row",
          "marginHorizontal": 16,
          "marginTop": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
          }
        }
        testID="back-button"
      >
        <Text />
      </View>
    </RCTSafeAreaView>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderTopLeftRadius": 56,
          "borderTopRightRadius": 56,
          "flex": 1,
          "marginTop": 240,
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          {
            "height": 300,
            "position": "absolute",
            "top": -150,
            "width": 300,
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": "100%",
              "resizeMode": "contain",
              "width": "100%",
            }
          }
        />
      </View>
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginTop": 160,
          }
        }
      >
        Bacterial Spot
      </Text>
      <Text
        style={
          {
            "marginHorizontal": 16,
            "marginTop": 16,
            "textAlign": "center",
          }
        }
      >
        Bacterial Spot is a common disease in plants.
      </Text>
      <View
        style={
          {
            "alignSelf": "flex-start",
            "marginBottom": 16,
            "marginHorizontal": 22,
            "paddingRight": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginTop": 16,
            }
          }
        >
          Crops:
        </Text>
        <Text
          style={
            {
              "marginLeft": 16,
            }
          }
        >
          Tomato, Pepper, Eggplant
        </Text>
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginTop": 16,
            }
          }
        >
          Symptoms:
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Leaf spots
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Wilting
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Fruit rot
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginTop": 16,
            }
          }
        >
          Causes:
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Bacteria
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Poor drainage
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginTop": 16,
            }
          }
        >
          Cures:
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Remove infected plants
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Use resistant varieties
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginTop": 16,
            }
          }
        >
          Preventive Measures:
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Proper irrigation
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginLeft": 16,
              "marginVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "marginRight": 8,
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "flex": 1,
              }
            }
          >
            Crop rotation
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
